src/main.cpp

```diff
@@ -302,11 +301,29 @@ void nimble_port_ll_task_func(void* args) {
 }
 }
 
+void calibrate_lf_clock_rc(nrf_drv_clock_evt_type_t event) {
+  // 16 * 0.25s = 4s calibration cycle
+  // Not recursive, call is deferred via internal calibration timer
+  nrf_drv_clock_calibration_start(16, calibrate_lf_clock_rc);
+}
+
 int main(void) {
   logger.Init();
 
   nrf_drv_clock_init();
 
+  // In some cases (eg. when a watchdog is enabled),
+  // the NRF SDK incorrectly assumes the LF clock is already running,
```

Comment:
Can you elaborate on this and the purpose of `while (!nrf_clock_lf_is_running()) {}` assuming it's related?

Reply:
The `nrf_drv_clock_calibration_start` requires the clock to be running, if it is not running the call fails and the calibration loop is not started. Starting the clock takes a bit of time in the background.

I'll move this check into the `#if (CLOCK_CONFIG_LF_SRC == NRF_CLOCK_LFCLK_RC)` guard, since it only concerns that case.